psalmist_frontend/src/components/pages/NewSong.js

```diff
@@ -2,63 +2,67 @@ import React, { useRef, useState, useEffect } from "react";
 import { connect, useDispatch } from "react-redux";
 import { Editor } from "@tinymce/tinymce-react";
 import PropTypes from "prop-types";
-import { addSong, getSong } from "../../actions/songActions";
+import { addSong, getSong, getSongs } from "../../actions/songActions";
 
 const NewSong = ({ song: { current, song }, addSong, getSong }) => {
   const dispatch = useDispatch();
 
-  useEffect(async () => {
-    try {
-      if (current) {
-        await dispatch(getSong(current));
```

Comment:
Why would you need to await a dispatch? Normally, you simply dispatch an action and the redux store will update. Whenever the store updates any subscribed component will get the updated value. In this case, it seems like you're getting a song. When the store updates, I suggest you listen for the updated song and then continue with what the component is suppose to do.